customcalendarview/src/main/java/com/stacktips/view/CustomCalendarView.java

```diff
@@ -0,0 +1,1010 @@
+            default:
+                childView = switchGetView1(index);
+                break;
+        }
+        return childView;
+    }
+
+    private Calendar getTodaysCalendar() {
+        Calendar currentCalendar = Calendar.getInstance(mLocale);
+        currentCalendar.setFirstDayOfWeek(getFirstDayOfWeek());
+        return currentCalendar;
+    }
+
+    /**
+     * refreshCalendar.
+     *
+     * @param currentCalendar is calendar.
+     */
+    public void refreshCalendar(Calendar currentCalendar) {
+        this.mCurrentCalendar = currentCalendar;
+        this.mCurrentCalendar.setFirstDayOfWeek(getFirstDayOfWeek());
+        mLocale = Locale.getDefault();
+        // Set date title
+        initializeTitleLayout();
+        // Set weeks days titles
+        initializeWeekLayout();
+        // Initialize and set days in calendar
+        setDaysInCalendar();
+    }
+
+    public int getFirstDayOfWeek() {
+        return firstDayOfWeek;
+    }
+
+    public void setFirstDayOfWeek(int firstDayOfWeek) {
+        this.firstDayOfWeek = firstDayOfWeek;
+    }
+
+    /**
+     * markDayAsCurrentDay.
+     *
+     * @param calendar is calendar.
+     */
+    public void markDayAsCurrentDay(Calendar calendar) {
+        if (calendar != null && CalendarUtils.isToday(calendar)) {
+            DayView dayOfMonth = getDayOfMonthText(calendar);
+            if (dayOfMonth == null) {
+                return;
+            }
+            dayOfMonth.setTextColor(currentDayOfMonth);
+        }
+    }
+
+    /**
+     * markDayAsSelectedDay.
+     *
+     * @param currentDate is currentdate.
+     *
+     */
+    public void markDayAsSelectedDay(Date currentDate) {
+        final Calendar currentCalendar = getTodaysCalendar();
+        currentCalendar.setFirstDayOfWeek(getFirstDayOfWeek());
+        currentCalendar.setTime(currentDate);
+        // Clear previous marks
+        clearDayOfTheMonthStyle(lastSelectedDay);
+        // Store current values as last values
+        storeLastValues(currentDate);
+        // Mark current day as selected
+        DayView view = getDayOfMonthText(currentCalendar);
+        if (view == null) {
+            return;
+        }
+        ShapeElement shapeElement = new ShapeElement();
+        shapeElement.setShape(ShapeElement.RECTANGLE);
+        shapeElement.setRgbColor(RgbColor.fromArgbInt(selectedDayBackground.getValue()));
+        view.setBackground(shapeElement);
+        view.setTextColor(selectedDayTextColor);
+    }
+
+    private void storeLastValues(Date currentDate) {
+        lastSelectedDay = currentDate;
+    }
+
+    public CalendarListener getCalendarListener() {
+        return calendarListener;
+    }
+
+    public void setCalendarListener(CalendarListener calendarListener) {
+        this.calendarListener = calendarListener;
+    }
+
+    private final ClickedListener onDayOfMonthClickListener = new ClickedListener() {
+        private Text switchOnClick2(String tagId, Component view) {
```

Comment:
change the naming convention to onClickExtended1 & onClickExtended2.Place the extended methods in order..